pkg/controller/unifiedpushserver/unifiedpushserver.go

```diff
@@ -0,0 +1,207 @@
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: cr.Name,
+					Containers: []corev1.Container{
+						{
+							Name:            "ups",
+							Image:           unifiedpush.image(),
+							ImagePullPolicy: corev1.PullAlways,
+							Env: []corev1.EnvVar{
+								{
+									Name:  "POSTGRES_SERVICE_HOST",
+									Value: fmt.Sprintf("%s-postgresql", cr.Name),
+								},
+								{
+									Name:  "POSTGRES_SERVICE_PORT",
+									Value: "5432",
+								},
+								{
+									Name: "POSTGRES_USER",
+									ValueFrom: &corev1.EnvVarSource{
+										SecretKeyRef: &corev1.SecretKeySelector{
+											Key: "database-user",
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: fmt.Sprintf("%s-postgresql", cr.Name),
+											},
+										},
+									},
+								},
+								{
+									Name: "POSTGRES_PASSWORD",
+									ValueFrom: &corev1.EnvVarSource{
+										SecretKeyRef: &corev1.SecretKeySelector{
+											Key: "database-password",
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: fmt.Sprintf("%s-postgresql", cr.Name),
+											},
+										},
+									},
+								},
+								{
+									Name: "POSTGRES_DATABASE",
+									ValueFrom: &corev1.EnvVarSource{
+										SecretKeyRef: &corev1.SecretKeySelector{
+											Key: "database-name",
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: fmt.Sprintf("%s-postgresql", cr.Name),
+											},
+										},
+									},
+								},
+							},
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          "ups",
+									Protocol:      corev1.ProtocolTCP,
+									ContainerPort: 8080,
+								},
+							},
+							ReadinessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/rest/applications",
+										Port: intstr.IntOrString{
+											Type:   intstr.Int,
+											IntVal: 8080,
+										},
+									},
+								},
+								InitialDelaySeconds: 15,
+								TimeoutSeconds:      2,
+							},
+						},
+						{
+							Name:            "ups-oauth-proxy",
+							Image:           proxy.image(),
+							ImagePullPolicy: corev1.PullAlways,
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          "public",
+									Protocol:      corev1.ProtocolTCP,
+									ContainerPort: 4180,
+								},
+							},
+							Args: []string{
+								"--provider=openshift",
+								fmt.Sprintf("--openshift-service-account=%s", cr.Name),
+								"--upstream=http://localhost:8080",
+								"--http-address=0.0.0.0:4180",
+								"--skip-auth-regex=/rest/sender,/rest/registry/device,/rest/prometheus/metrics,/rest/auth/config",
+								"--https-address=",
+								"--cookie-secret=SECRET",
```

Comment:
The value here should be generated, rather than the hardcoded "SECRET" from the example. The [docs](https://github.com/openshift/oauth-proxy#configuration) suggest using something equivalent to `python -c 'import os,base64; print base64.b64encode(os.urandom(16))'`.